301-custom-images-at-scale/download.json

```diff
@@ -0,0 +1,40 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string"
+    },
+    "uniquePrefix": {
+      "type": "string"
+    },
+    "imageLocation": {
+      "type": "string"
+    },
+    "imageKey": {
+      "type": "string"
+    },
+    "downloadScriptURI": {
+      "type": "string"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat('transfer', parameters('uniquePrefix'), 'vm/ext')]",
+      "apiVersion": "2015-05-01-preview",
+      "location": "[parameters('location')]",
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.2",
+        "settings": {
+          "fileUris": [
+            "[parameters('downloadScriptURI')]"
+          ],
+          "commandToExecute": "[concat('bash download.sh ', parameters('imageLocation'),' ', parameters('imageKey'))]"
```

Comment:
Pass this as part of protectedSettings and use handlerVersion 1.4 since keys are sensitive